pages/_app.test.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { FC, useEffect } from 'react'
+import Cookies from 'js-cookie'
+import { mount } from 'enzyme'
+import App from './_app'
+import { UserContext, UserContextProps, UserState, USER_STATE_DEFAULT } from '../context/UserContext'
+import { COOKIE_NAME_ADULT_FILTER, COOKIE_NAME_NEW_TAB, COOKIE_NAME_SEARCH_COUNT } from '../helpers/_cookies'
+
+describe('App', () => {
+  let ContextWrapper
+  let TestComponent: FC<UserContextProps>
+  let cookies = {}
+
+  function setCookies(_cookies) {
+    for (const key in _cookies) {
+      Cookies.set(key, _cookies[key])
+    }
+    cookies = _cookies
+  }
+
+  function resetCookies() {
+    for (const key in cookies) {
+      Cookies.remove(key)
+    }
+    cookies = {}
+  }
+
+  beforeEach(() => {
+    ContextWrapper = () => {
+      return (
+        <UserContext.Consumer>
+          {({ userState, setUserState }) => <TestComponent userState={userState} setUserState={setUserState} />}
+        </UserContext.Consumer>
+      )
+    }
+  })
+
+  afterEach(() => {
+    resetCookies()
+  })
+
+  it('should pass default user state to context', function () {
+    TestComponent = () => null
+
+    const wrapper = mount(<App Component={ContextWrapper} pageProps={null} router={null} />)
+
+    expect(wrapper.find(TestComponent).prop('userState')).toEqual(USER_STATE_DEFAULT)
+  })
+
+  it('should pass user state merged from cookies to context', function () {
+    setCookies({
+      [COOKIE_NAME_SEARCH_COUNT]: 3,
+      [COOKIE_NAME_NEW_TAB]: true,
+      [COOKIE_NAME_ADULT_FILTER]: 0, // deliberately included 0 as an edge case here (falsy)
```

Comment:
0 is actually a correct value. Previous implementation was a mistake.
we have 3 filter which values are 0, 1, 2

Reply:
Yes, that's why I check for `!== undefined` in `mergeCookiesWithUserState`.

Reply:
ahh! i get the comment now. Ook cool, thanks